wagtail_newsletter/actions.py

```diff
@@ -0,0 +1,31 @@
+from typing import cast
+
+from wagtail.admin import messages
+
+from . import campaign_backends
+from .models import NewsletterPageMixin
+
+
+def save_campaign(request, page: NewsletterPageMixin):
```

Comment:
maybe a `-> None`, so that we can instantly understand that this function returns nothing, without having to skim through it to get this information? :slightly_smiling_face:

Reply:
Also... As this seems to be the "business logic" layer, I think that according to the MVC (or MTV, in Django's own terminology :shrug: ) architecture it shouldn't have a dependency to the HTTP layer?  

It would probably be better to return a "pure data" result here, that the View can then convert into action on the HTTP layer?

Reply:
I think of `actions.py` as part of the view layer. I've extracted the logic here from `wagtail_hooks.py` because it was getting too large.